pkg/storage/fs/cback/utilities.go

```diff
@@ -0,0 +1,360 @@
+package cback
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/errtypes"
+)
+
+type backUpResponse struct {
+	Detail    string `json:"detail"`
+	ID        int    `json:"id"`
+	Name      string `json:"name"`
+	Source    string `json:"source"`
+	Substring string //Used in function
+}
+
+type snapshotResponse struct {
+	Detail string   `json:"detail"`
+	ID     string   `json:"id"`
+	Time   string   `json:"time"`
+	Paths  []string `json:"paths"`
+}
+
+type contents struct {
+	Name   string  `json:"name"`
+	Type   string  `json:"type"`
+	Mode   uint64  `json:"mode"`
+	Mtime  float64 `json:"mtime"`
+	Atime  float64 `json:"atime"`
+	Ctime  float64 `json:"ctime"`
+	Inode  uint64  `json:"inode"`
+	Size   uint64  `json:"size"`
+	Detail string  `json:"detail"`
+}
+
+type fsReturn struct {
+	Type   provider.ResourceType
+	Mtime  uint64
+	Size   uint64
+	Path   string
+	Detail string
+}
+
+var permID = provider.ResourcePermissions{
+	AddGrant:             false,
+	CreateContainer:      false,
+	Delete:               false,
+	GetPath:              true,
+	GetQuota:             true,
+	InitiateFileDownload: true,
+	InitiateFileUpload:   false,
+	ListGrants:           true,
+	ListContainer:        true,
+	ListFileVersions:     true,
+	ListRecycle:          false,
+	Move:                 false,
+	RemoveGrant:          false,
+	PurgeRecycle:         false,
+	RestoreFileVersion:   true,
+	RestoreRecycleItem:   false,
+	Stat:                 true,
+	UpdateGrant:          false,
+	DenyGrant:            false}
+
+var checkSum = provider.ResourceChecksum{
+	Sum:  "",
+	Type: provider.ResourceChecksumType_RESOURCE_CHECKSUM_TYPE_UNSET,
+}
+
+func mapReturn(fileType string) (provider.ResourceType, error) {
+	/* This function can be changed accordingly, depending on the file type
+	being return by the APIs */
+
+	switch fileType {
+	case "file":
+		return provider.ResourceType_RESOURCE_TYPE_FILE, nil
+
+	case "dir":
+		return provider.ResourceType_RESOURCE_TYPE_CONTAINER, nil
+
+	default:
+		return provider.ResourceType_RESOURCE_TYPE_INVALID, errtypes.NotFound("Resource type unrecognized")
+	}
+}
+
+func (fs *cback) getRequest(userName, url string, reqType string, body io.Reader) (io.ReadCloser, error) {
+
+	req, err := http.NewRequest(reqType, url, body)
+	req.SetBasicAuth(userName, fs.conf.ImpersonatorToken)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if body != nil {
+		req.Header.Add("Content-Type", "application/json")
+	}
+
+	req.Header.Add("accept", `application/json`)
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == 404 {
```

Comment:
for all these if cases, use a switch statement, and instead of using directly the numbers, use the const values defined in the http library (404 -> http.StatusNotFound)

Before check for a of status (status >= 200 && status < 300).

Reply:
Checking if the code is NOT an Okay code (resp.StatusCode > 200 && resp.StatusCode >= 300), if this is so the switch statement is invoked which returns a different error according to the code. If the return status is OKAY, it will simply return the body, and nil error.